clang-tools-extra/clang-tidy/tool/run-clang-tidy.py

```diff
@@ -446,6 +448,12 @@ async def main() -> None:
         action="store_true",
         help="Allow empty enabled checks.",
     )
+    parser.add_argument(
+        "-ignore-insert-conflict",
```

Comment:
This option is currently a no-op as it is, because it doesn't change anything. The default value is True, and this option can only set it to True. If anything, there should be a `-no-ignore-insert-conflict` option that sets it to False, given a default of True.

I still don't fully understand what problem this patch is trying to solve, could you ellaborate? Maybe this patch is not needed at all?

Reply:
The problem is that `-ignore-insert-conflict` does not eist to older versions of clang to 15.0, for example I am using 14.0

Reply:
Ok, then I see the following options:

- If possible, bump your installation to clang 15 (we are about to release clang 20, so it's fairly old ;) )
- Add a `-no-ignore-insert-conflict` option.
- Automatically check if the `clang_apply_replacements_binary` supports the `-ignore-insert-conflict` flag. You can easily do that by running the `clang-apply-replacements --help` command and checking if the output contains the flag. This is probably the most convenient option for the users since no changes to how the program is run are needed.

Reply:
Sorry if I use an old version of clang, the system is configured by my company and I cannot chose. Anyway, I will check your third option once I have time. Let me some time.

Reply:
Should be Okay now.